cpmpy/transformations/linearize.py

```diff
@@ -550,7 +558,8 @@ def canonical_comparison(lst_of_expr):
 
         elif isinstance(cpm_expr, Comparison):
             lhs, rhs = cpm_expr.args
-            if isinstance(lhs, Comparison) and cpm_expr.name == "==":  # reification of comparison
+            if isinstance(lhs, Comparison) and (is_bool(rhs) or isinstance(rhs, Expression) and rhs.is_bool()):
```

Comment:
So we need both an `is_bool(rhs)` en `rhs.is_bool()`? There is some subtle difference between the two?

Reply:
Yes, so `is_bool` checks whether it is a Boolean constant (also non-cpmpy ones). `.is_bool()` checks whether a CPMpy expression is of Boolean type.